v2/tools/generator/internal/codegen/pipeline/report_resource_versions.go

```diff
@@ -153,35 +168,62 @@ func (report *ResourceVersionsReport) summarize(definitions astmodel.TypeDefinit
 	}
 }
 
-// WriteTo creates a file containing the generated report
-func (report *ResourceVersionsReport) WriteTo(outputFile string) error {
+// SaveAllResourcesReportTo creates a file containing the generated report
+func (report *ResourceVersionsReport) SaveAllResourcesReportTo(outputFile string) error {
 
 	klog.V(1).Infof("Writing report to %s", outputFile)
-	report.frontMatter = report.readFrontMatter(outputFile)
+	frontMatter := report.readFrontMatter(outputFile)
 
 	var buffer strings.Builder
-	err := report.WriteToBuffer(&buffer)
+	err := report.WriteAllResourcesReportToBuffer(frontMatter, &buffer)
 	if err != nil {
 		return errors.Wrapf(err, "writing versions report to %s", outputFile)
 	}
 
-	if _, err := os.Stat(outputFile); os.IsNotExist(err) {
+	outputFolder := filepath.Dir(outputFile)
+	if _, err := os.Stat(outputFolder); os.IsNotExist(err) {
+		err = os.MkdirAll(outputFolder, 0o700)
+		if err != nil {
+			return errors.Wrapf(err, "Unable to create directory %q", outputFile)
+		}
+	}
+
+	return os.WriteFile(outputFile, []byte(buffer.String()), 0o600)
+}
+
+// SaveGroupResourcesReportTo creates a file containing the generated report
```

Comment:
minor: Improve the doc comments here and above to explain the difference between the `AllResourcesReport` and the `GroupResourcesReport`?

Reply:
Done.